replaying-share/src/test/java/com/jakewharton/rx/ReplayingShareFlowableTest.java

```diff
@@ -242,4 +243,29 @@
     replayed.subscribe(observer3);
     observer3.assertValues("initB");
   }
+
+  @Test public void unsubscribeInOnSubscribePreventsCacheEmission() {
+    PublishProcessor<String> upstream = PublishProcessor.create();
+    Flowable<String> replayed = upstream.compose(ReplayingShare.<String>instance());
+    replayed.subscribe();
+    upstream.onNext("something to cache");
+
+    TestSubscriber<String> testSubscriber = new TestSubscriber<>(new Subscriber<String>() {
+      @Override
+      public void onSubscribe(Subscription subscription) {
+        subscription.cancel();
```

Comment:
You can call `testSubscriber.dispose()` (or `cancel()`) before calling `subscribe(testSubscriber)` to achieve this effect.

Reply:
Done